gpytorch/module.py

```diff
@@ -138,9 +138,14 @@ def named_added_loss_terms(self):
         return _extract_named_added_loss_terms(module=self, memo=None, prefix="")
 
     def named_hyperparameters(self):
+        from .variational._variational_distribution import _VariationalDistribution
+
         for name, param in self.named_parameters():
-            if "variational_" not in name:
-                yield name, param
+            gen = self._named_members(
```

Comment:
What if there are several `_VariationalDistribution` modules? This will give the wrong answer silently. At least, the `for elem in gen:` line and the one below should be indented 4 spaces to the right. I'd even go further, and remove all non-public `nn.Module` methods:

```python
    for module_prefix, module in self.named_modules():
        if not isinstance(module, _VariationalDistribution):
            for elem in module.named_parameters(prefix=module_prefix, recurse=False):
                yield elem
```